gloo/transport/tcp/unbound_buffer.cc

```diff
@@ -72,11 +84,12 @@ void UnboundBuffer::waitRecv(int* rank, std::chrono::milliseconds timeout) {
                   "ms for recv operation to complete"));
     }
   }
-
+  auto recvCompleted = recvCompletions_ > 0;
```

Comment:
Can't understand this state, `recvCompleted`. Mind adding comments to explain it a bit?

Reply:
Sure, I'll add some comments.

The goal here is to return `true` if the receive completed (i.e. we got data from the remote side) and `false` if it was aborted. One way to know that we got data from the remote side is if there was at least one `recvCompletion`, so I used this to determine the return value. We could potentially also use `!abortWaitRecv`, but we could run into an issue where the receive completes successfully, then a thread sets `abortWaitRecv` before we return out of `waitRecv`, in which case returning `!abortWaitRecv` would not be correct

Reply:
Will this happen? 
Before it returns out of `waitRecv`, doesn't it hold the lock?

Reply:
That makes sense, the lock would be reacquired once the thread is awakened. This should be changed to

`if (abortWaitSend_): return false`
else, decrement `recvCompletions` as below and return true